src/components/lifecycle/lifecycle.ts

```diff
@@ -0,0 +1,110 @@
+import { LitElement, html } from 'lit';
+import { customElement, property } from 'lit/decorators.js';
+import Styles from './lifecycle.scss';
+import '../button/button';
+import { BUTTON_SIZES } from '../button/defs';
+
+@customElement('kd-lifecycle')
+export class LitLifecycle extends LitElement {
+  @property({ type: Number }) counter = 0;
+
+  /** Specifies the size of the button. */
+  @property({ type: String })
+  size: BUTTON_SIZES = BUTTON_SIZES.SMALL;
+
+  @property({ type: String })
+  primaryBtnText = '';
+
+  @property({ type: String })
+  secondaryBtnText = '';
+
+  static override styles = Styles;
+  constructor() {
+    super();
+    console.log('Constructor: Component is being created');
+    this.counter = 0; // Initial property value
+  }
+
+  override connectedCallback() {
+    super.connectedCallback();
+    console.log('ConnectedCallback: Component is added to the DOM');
+    //set up additional listeners or logic here
+  }
+
+  override disconnectedCallback() {
+    console.log('DisconnectedCallback: Component is removed from the DOM');
+    //Clean up resources or listeners here
+    super.disconnectedCallback();
+  }
+
+  override shouldUpdate(
+    changedProperties: Map<string | number | symbol, unknown>
+  ): boolean {
+    console.log('ShouldUpdate: Check if the component should update');
+    return super.shouldUpdate(changedProperties);
+    // Implement logic to decide if the component should update
+    // Return true to update, false to skip the update
+  }
+
+  override willUpdate(
+    changedProperties: Map<string | number | symbol, unknown>
+  ) {
+    console.log('WillUpdate: Before the component updates');
+    super.willUpdate(changedProperties);
+    // Implement logic to run before the update
+  }
+
+  override update(changedProperties: Map<string | number | symbol, unknown>) {
+    console.log('Update: During the update process');
+    super.update(changedProperties);
+    // Optionally request updates or perform other tasks here
+  }
+
+  override render() {
+    console.log('Render: Rendering the component');
+    return html`
+      <div>
+        <h2>Lifecycle Component</h2>
```

Comment:
Avoid using heading tags inside reusable components, as they won't be aware of the document hierarchy.